diff --git a/node_modules/@ethersproject/providers/lib/json-rpc-batch-provider.js b/node_modules/@ethersproject/providers/lib/json-rpc-batch-provider.js
index 20f35c5..c495957 100644
--- a/node_modules/@ethersproject/providers/lib/json-rpc-batch-provider.js
+++ b/node_modules/@ethersproject/providers/lib/json-rpc-batch-provider.js
@@ -68,14 +68,14 @@ var JsonRpcBatchProvider = /** @class */ (function (_super) {
                     // on whether it was a success or error
                     batch.forEach(function (inflightRequest, index) {
                         var payload = result[index];
-                        if (payload.error) {
-                            var error = new Error(payload.error.message);
-                            error.code = payload.error.code;
-                            error.data = payload.error.data;
+                        if (payload && payload?.error) {
+                            var error = new Error(payload?.error.message);
+                            error.code = payload?.error.code;
+                            error.data = payload?.error.data;
                             inflightRequest.reject(error);
                         }
                         else {
-                            inflightRequest.resolve(payload.result);
+                            inflightRequest.resolve(payload?.result);
                         }
                     });
                 }, function (error) {
